pkg/pillar/cmd/zedrouter/flowstats.go

```diff
@@ -654,7 +664,107 @@ func DNSStopMonitor(bnNum int) {
 	}
 }
 
-func checkDNSPacketInfo(bnNum int, packet gopacket.Packet) {
+// Monitor the dhcp packets for switched network instance
+func checkDHCPPacketInfo(bnNum int, packet gopacket.Packet, ctx *zedrouterContext) {
+	var isIPv4, isReplyAck, needUpdate, foundDstMac bool
+	var vifInfo []types.VifNameMac
+	var netstatus types.NetworkInstanceStatus
+
+	// use the IPAssigments of the NetworkInstanceStatus, since this is switched net
+	// and the field will not be assigned or modified by others
+	pub := ctx.pubNetworkInstanceStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		netstatus = cast.CastNetworkInstanceStatus(st)
+		if netstatus.Type != types.NetworkInstanceTypeSwitch || netstatus.BridgeNum != bnNum {
+			continue
+		}
+		vifInfo = netstatus.Vifs
+		break
+	}
+	if len(vifInfo) == 0 { // there is no Mac on the bridge
+		return
+	}
+
+	etherLayer := packet.Layer(layers.LayerTypeEthernet)
+	if etherLayer != nil {
+		etherPkt, _ := etherLayer.(*layers.Ethernet)
+		for _, vif := range vifInfo {
+			if strings.Compare(etherPkt.DstMAC.String(), vif.MacAddr) == 0 {
+				foundDstMac = true
+				break
+			}
+		}
+	}
+	if foundDstMac == false { // dhcp packet not for this bridge App ports
```

Comment:
I find the style of comparing booleans against true/false error prone.
For example, could have been expressed as if foundDstMac != true.

The most concise and clear is
 if !foundDstMac {

Reply:
ok